src/adapter/src/coord/ddl.rs

```diff
@@ -503,12 +529,39 @@ impl Coordinator {
         let conn = conn_id.map(|id| active_conns.get(id).expect("connection must exist"));
 
         let TransactionResult {
-            builtin_table_updates,
+            mut builtin_table_updates,
             audit_events,
         } = catalog
             .transact(Some(&mut *controller.storage), oracle_write_ts, conn, ops)
             .await?;
 
+        // The `mz_ssh_tunnel_connections` table is weird. It's contents are not fully derived from
+        // catalog state, they're derived from the secrets controller. However, it still must be
+        // kept in-sync with the catalog state. Storing the contents of the table in the durable
+        // catalog would simplify the situation, but then we'd have to somehow encode public keys
+        // the CREATE SQL of connections, which is annoying. In order to successfully balance all of
+        // these constraints, we handle all updates to the table separately.
+        for (ssh_conn, diff) in ssh_conn_to_update {
+            let secret = self.secrets_controller.reader().read(ssh_conn).await?;
```

Comment:
Eek, this is happening on the main coordinator thread, right? That's a bit scary.

Reply:
I had an idea for how to remove this for `CREATE` connection without removing any consistency guarantees and just pushed that in b4762ad1e18c55644b3efcc1f6f7bba2c6225d2f. In a way it may make life harder in the future for whoever tries to remove secret controller calls from the main coord thread, but for now I think we get the best of both worlds. Let me know what you think.

I still plan on doing the discussed async task work for dropping SSH connections since that is still a new blocking call.